sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -119,7 +123,7 @@
             analyzeSentimentResults.add(convertToAnalyzeSentimentResult(documentSentiment));
         }
         for (DocumentError documentError : sentimentResponse.getErrors()) {
-            analyzeSentimentResults.add(new AnalyzeSentimentResult(documentError.getId(), null,
+            analyzeSentimentResults.add(new AnalyzeSentimentResultImpl(documentError.getId(), null,
```

Comment:
Not so sure, but I see that `AnalyzeSentimentResultImpl` isn't a generated class but we still have it under `implementation.models` package?

Reply:
Is that the new design change so that we avoid having public constructors in our model?

Reply:
Yes. It is. We explore AnalyzeSentimentResult interface to user but hide the implementation of it in implementation folder.

Reply:
interface approach is no longer a option. keep as original design, adding overload constructor and deprecate constructor as needed.